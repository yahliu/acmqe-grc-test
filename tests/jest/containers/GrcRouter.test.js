/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import GrcRouter from '../../../src-web/containers/GrcRouter'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom'
import * as reducers from '../../../src-web/reducers'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import GrcApolloClient from '../../../lib/client/apollo-client'
import { ApolloProvider } from 'react-apollo'
import { Provider } from 'react-redux'
import { ROLES } from '../../../lib/shared/constants'
import toJson from 'enzyme-to-json'

describe('Grc Router container test', () => {
  it('renders as expected', () => {
    const preloadedState = window.__PRELOADED_STATE__
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), preloadedState, composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const match = {
      'path': '/multicloud',
      'url': '/multicloud',
      'isExact': false,
      'params': {}
    }
    const component = shallow(
      <ApolloProvider client={GrcApolloClient.getGrcClient()}>
        <Provider store={store}>
          <BrowserRouter>
            <GrcRouter
              match={match}
              ROLES={ROLES}
            />
          </BrowserRouter>
        </Provider>
      </ApolloProvider>
    )
    expect(toJson(component)).toMatchSnapshot()
  })
})
