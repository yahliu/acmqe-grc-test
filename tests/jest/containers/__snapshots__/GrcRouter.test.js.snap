// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grc Router container test renders as expected 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
